src/main/features/claim/routes/claimant-add.ts

```diff
@@ -0,0 +1,59 @@
+import * as express from 'express'
+import { Paths } from 'claim/paths'
+
+import { Form, FormValidationError } from 'app/forms/form'
+import { FormValidator } from 'app/forms/validation/formValidator'
+import ErrorHandling from 'common/errorHandling'
+import { YesNo } from 'app/forms/models/yesNo'
+
+import { ClaimDraftMiddleware } from 'claim/draft/claimDraftMiddleware'
+import { ValidationError } from 'class-validator'
+import { ClaimantAddition } from 'app/forms/models/claimantAddition'
+import { Claimants } from 'common/router/claimants'
+
+const MAX_CLAIMANTS_ALLOWED: number = 20
+const ERROR_MESSAGE: string = `You can't add more than ${MAX_CLAIMANTS_ALLOWED} claimants`
+
+function renderView (form: Form<ClaimantAddition>, res: express.Response) {
+  const claimants = res.locals.user.legalClaimDraft.claimants
+
+  res.render(Paths.claimantAdditionPage.associatedView, {
+    form: form,
+    claimants: res.locals.user.viewDraft.isClaimantDeleted || claimants.length > 1 ? claimants : null,
+    maxAllowedLimit: MAX_CLAIMANTS_ALLOWED
+  })
+}
+
+let addErrorMessage = function (form: Form<ClaimantAddition>) {
+  const validationError = new ValidationError()
+  validationError.property = 'isAddClaimant'
+  validationError.target = { 'isAddClaimant': 'YES' }
+  validationError.value = 'YES'
+  validationError.constraints = { ['isAddClaimant']: ERROR_MESSAGE }
+  form.errors.push(new FormValidationError(validationError, ''))
+}
+export default express.Router()
+  .get(Paths.claimantAdditionPage.uri, (req: express.Request, res: express.Response) => {
+    renderView(new Form(new ClaimantAddition()), res)
+  })
+  .post(Paths.claimantAdditionPage.uri, FormValidator.requestHandler(ClaimantAddition, ClaimantAddition.fromObject),
+    ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+      const form: Form<ClaimantAddition> = req.body
+
+      if (form.model.isAddClaimant === YesNo.YES && res.locals.user.legalClaimDraft.claimants.length === MAX_CLAIMANTS_ALLOWED) {
+        addErrorMessage(form)
+      }
+
+      if (form.hasErrors()) {
+        renderView(form, res)
+      } else {
+        if (form.model.isAddClaimant === YesNo.YES) {
+          Claimants.addClaimant(res)
+          await ClaimDraftMiddleware.save(res, next)
+          res.redirect(Paths.claimantTypePage.uri)
+        } else {
```

Comment:
I think you might need to save claimant in draft-store here as well. Or probably you don't need `Claimants.addClaimant(res)` on line 51 as well. I think this page is just a yes/no question.

Reply:
Claimants.addClaimant(res) adds empty Claimant object which is not required in case of no

Reply:
Help me understand why we need to add empty object, if it is due to loop index, we can use `loop.index0: the current iteration of the loop (0 indexed)` rather than `loop.index: the current iteration of the loop (1 indexed)`